zerver/tornado/event_queue.py

```diff
@@ -647,9 +646,9 @@ def missedmessage_hook(user_profile_id: int, client: ClientDescriptor, last_for_
         mentioned = 'mentioned' in flags and 'read' not in flags
         private_message = event['message']['type'] == 'private'
         # stream_push_notify is set in process_message_event.
-        stream_push_notify = event.get('stream_push_notify', False)
-        stream_email_notify = event.get('stream_email_notify', False)
-        wildcard_mention_notify = (event.get('wildcard_mention_notify', False) and
+        stream_push_notify = event.get('internal_data', {}).get('stream_push_notify', False)
+        stream_email_notify = event.get('internal_data', {}).get('stream_email_notify', False)
+        wildcard_mention_notify = (event.get('internal_data', {}).get('wildcard_mention_notify', False) and
```

Comment:
Do we need the `.get()` for accessing `internal_data`?  I would have thought the logic below would guarantee there's an `internal_data` field.  And actually the same for `stream_push_notify` and friends -- should these lines be just:
```
stream_push_notify = event['internal_data']['stream_push_notify']
```

?

Reply:
`internal_data` is only created/initialized if  `private`, `mentioned` and `wildcard_mentioned` flags are there whereas we test missedmessage_hook for normal stream message also in which case there is no `internal_data` in message event.